### PR TITLE
Update guide to UDFs with notes about `Series.applymap` deprecation and related changes

### PR DESCRIPTION
**Pull request by brandon-b-miller**
This PR updates our guide to UDFs notebook in the following ways:

- Notes deprecation of `cudf.Series.applymap`
- reorder sections such that `cudf.Series.apply` and `cudf.DataFrame.apply` are encountered before `applymap`, `apply_rows` and `apply_chunks`
- Adds recommendations about use cases for each of the above APIs to distinguish them
- Minor updates all over to better describe current state

EDIT: decided to just remove the docs for `applymap` at this point.

### CODE REVIEW

**Comment by brandon-b-miller** [comment]
rerun tests

File: docs/cudf/source/user_guide/guide-to-udfs.ipynb [hunk 1]
```diff
@@ -7,13 +7,24 @@
     "# Overview of User Defined Functions with cuDF"
```
Comment (shwina) [inline docs/cudf/source/user_guide/guide-to-udfs.ipynb:7]: <p>While it's good to do this in tests, maybe we don't want to compare the invocations of <code>apply()</code> with Pandas -- unless there's a difference we want to call out.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/10607/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (shwina) [inline docs/cudf/source/user_guide/guide-to-udfs.ipynb:7]: <blockquote>Functions used with <code>cudf.Series.apply</code> and <code>cudf.DataFrame.apply</code> can be expected to handle nulls using the same rules as the rest of cuDF. In most cases this translates to nulls propagating through unary and binary operations and yielding more nulls.</blockquote><p> </p><p>I think we could simplify while also being a bit more explicit here:</p><p> </p><p>The null value <code>NA</code> an propagates through unary and binary operations. Thus, <code>NA + 1</code>,<code>abs(NA)</code>, and <code>NA == NA</code> all return <code>NA</code>. To make this concrete, let's look at the same example from above, this time using nullable data:</p><p> </p><p>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/10607/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (shwina) [inline docs/cudf/source/user_guide/guide-to-udfs.ipynb:7]: <p>In other parts of this notebook, we use the second person "you", whereas here we use "the user". Let's be consistent -- I have a strong preference for the second person "you".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/10607/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (shwina) [inline docs/cudf/source/user_guide/guide-to-udfs.ipynb:7]: <p> </p><blockquote>Many problems in data science and engineering are well studied and there exist known parallel algorithms for making some desired transformation to some data. Many have corresponding CUDA solutions that may not exist as column level API in cuDF. To expose the ability to use these custom kernels, cuDF supports directly using custom cuda kernels written using <code>numba</code> on cuDF <code>Series</code> objects. In short, this means that if a user has knowledge of how to write a CUDA kernel in numba, they may simply pass cuDF <code>Series</code> objects to that kernel as if they were numba device arrays. Let's look at a basic example of how to do this.</blockquote><p> </p><p>Two observations here:</p><p> </p><ul><li>We should avoid categorizing users as coming from a data science or engineering background</li><li>It takes a bit of reading before I understand what this section is about.</li></ul><p> </p><p>I think we should be more direct here; something along the lines of:</p><p> </p><blockquote>In addition to the <code>Series.apply()</code> method for performing custom operations, you can also pass <code>Series</code> objects directly into [CUDA kernels written with Numba](https://numba.pydata.org/numba-doc/latest/cuda/kernels.html).</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/10607/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (shwina) [inline docs/cudf/source/user_guide/guide-to-udfs.ipynb:7]: <p>&gt;we just need to pre-allocate an output array and leverage the <code>forall</code> method mentioned above</p><p> </p><p>I can't remember where I first learned about avoiding the 'J' word <em>just</em> - but it was probably [here](https://software-carpentry.org/blog/2015/03/teaching-tips.html).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/10607/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (shwina) [inline docs/cudf/source/user_guide/guide-to-udfs.ipynb:7]: <blockquote>This example is adapted from cuDF's <a href="https://docs.rapids.ai/api/cudf/stable/api.html#cudf.core.dataframe.DataFrame.rolling" target="_blank">API documentation</a>. </blockquote><p> </p><p>Orthogonal to this PR, but the link here is broken. Ditto below with apply_grouped.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cudf/pull/10607/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='75'/>

Comment (brandon-b-miller) [reply docs/cudf/source/user_guide/guide-to-udfs.ipynb:7]: I fixed this link but actually I think the docs for `apply_grouped` are missing as I dont see them anywhere in our stable docs. Looking into this separately.

**Review by shwina — APPROVED** [review]

**Pull request merged**